Source/items.cpp

```diff
@@ -4664,8 +4655,10 @@ static void SpawnOnePremium(int i, int plvl, int myplr)
 		GetItemBonus(0, itype, plvl >> 1, plvl, true, !gbIsHellfire);
 
 		if (!gbIsHellfire) {
-			if (items[0]._iIvalue > 140000)
+			if (items[0]._iIvalue > 140000) {
+				keepgoing = true; // prevent breaking the do/while loop too early by failing hellfire's condition in while

```

Comment:
while this is a relatively ugly hack, I really think we should stop refactoring till we find and fix 1.2.0 issues :D

Reply:
Are you saying that to get us to focus on looking in to the bug? We are not going to include the refactoring in 1.2.1 if that's your concern.

Reply:
oh, didn't know :D